src/content/reference/react/useSyncExternalStore.md

```diff
@@ -397,7 +397,33 @@ function getSnapshot() {
 }
 ```
 
-If your store data is mutable, your `getSnapshot` function should return an immutable snapshot of it. This means it *does* need to create new objects, but it shouldn't do this for every single call. Instead, it should store the last calculated snapshot, and return the same snapshot as the last time if the data in the store has not changed. How you determine whether mutable data has changed depends on your mutable store.
+If the data in your store is mutable, your `getSnapshot` function should return an immutable snapshot of it. This means it *does* need to create new objects, but it shouldn't do this for every single call. Instead, it should cache the last calculated snapshot, and return a new object only if the data in the store has changed. If it has not changed, you should return the cached object. How you determine whether the data in your store has changed is specific to the store you're using.
+
+<DeepDive>
+
+#### Why does React re-render even if the values of your object have not changed? {/*a*/}
+
+JavaScript has a syntax called [object literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer), which are key-value pairs enclosed in curly braces (`{}`). This syntax creates a new object.
+
+However, when comparing objects, JavaScript only checks if the two objects are the same object, not if their *contents* are equal. Because the object literal syntax creates a new object, comparing two objects created with the object literal syntax always gives `false`, even when the values are the same. For example:
+
+```js
+object1 = {
+  name: "John",
+  favoriteNumber: 42
+};
+
+object2 = {
+  name: "John",
+  favoriteNumber: 42
+};
+
+console.log(object1 == object2); // false
+```
```

Comment:
Something is missing here, that connects it back to why `return {}` is a new object every time, something like this:

```suggestion
This also applies to the return values of functions:

```js
function getUser() {
  return {
    name: "John",
    favoriteNumber: 42
  };
}

const object1 = getUser();
const object2 = getUser();

console.log(object1 == object2); // false
```

And maybe add an example of hosting the object out to show how it doesn't change?
```

Reply:
The examples would be better as chat settings like in the explanations above rather than users.

Reply:
@rickhanlonii I'm not sure how I should change the code example. I want it to be clear to the reader that this isn't specific to `getSnapshot` handlers (we include examples of that above), but it needs to include returning an object from a function (like you said).

We also want to make it clear that the values of the retuned object are the same, but then it doesn't make sense why you would be comparing the returned values.

The best I can think of is something like this:

```js
function getMathConstants() {
  return {
    pi: 3.14159,
    e: 2.71828
  }
}

let mathConstants1 = getMathConstants();
let mathConstants2 = getMathConstants();

console.log(mathConstants1 == mathConstants2);
// false
```

But that makes very little sense. And, like you said, it should probably be related to the example chat app.

Any suggestions?

Reply:
ping @rickhanlonii